src/utils/supertype.rs

```diff
@@ -186,6 +187,15 @@ pub fn get_supertype(l: &DataType, r: &DataType) -> Option<DataType> {
             //TODO(sammy): add time, struct related dtypes
             (Boolean, Float32) => Some(Float32),
             (Boolean, Float64) => Some(Float64),
+            // TODO(Clark): Add support for getting supertype for two fixed size lists once Arrow2 supports such a cast.
+            // (FixedSizeList(inner_left_field, inner_left_size), FixedSizeList(inner_right_field, inner_right_size)) if inner_left_size == inner_right_size => {
+            //     let inner_st = inner(&inner_left_field.dtype, &inner_right_field.dtype)?;
+            //     Some(DataType::FixedSizeList(Box::new(Field::new(inner_left_field.name.clone(), inner_st)), *inner_left_size))
+            // }
+            (FixedSizeList(inner_left_field, inner_left_size), List(inner_right_field)) => {
+                let inner_st = inner(&inner_left_field.dtype, &inner_right_field.dtype)?;
```

Comment:
I think you want to call the top level function rather than `inner` since that handles both the `(l,r)` and `(r, l)` case for dtypes.

Reply:
Good point!